account_invoice_analytic_search/models/account_invoice.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014-2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = "account.invoice"
+
+    @api.multi
+    @api.depends('invoice_line_ids')
```

Comment:
@nikul-serpentcs You'd better depends on 'invoice_line_ids.account_analytic_id' as all recordsets will be fetched with these before entering loop.